docfx_yaml/extension.py

```diff
@@ -720,23 +720,45 @@ def sanitize_uidname_field(toc_yaml):
                 sanitize_uidname_field(module['items'])
             module.pop('uidname')
 
-    # Parses the package name and returns package name and module name.
-    def find_package_name(package_name):
+    # Parses the package name and returns unique identifer and name.
+    def find_unique_name(package_name, entries):
         for name in package_name:
-            if name != "google" and name != "cloud":
+            # Only find unique identifiers beside "google" and "cloud"
+            # For example, if given
+            #   "google.cloud.spanner.v1.params_v1.types"
+            #   "google.cloud.spanner.v1.instance_v1.types"
+            # it will return "instace_v1" or "params_v1" and "types".
+            if name != "google" and name != "cloud" and entries[name] == 1:
                 return [name, package_name[-1]]
 
     # Used to disambiguate names that have same entries.
     def disambiguate_toc_name(toc_yaml):
         names = {}
```

Comment:
Consider a different name from `names`, as this holds the number of times each name occurs, not just the names. `name_counts`?

Reply:
changed it to `name_entries` since we call its list `entries` above in the `find_package_name` function!